packit_service/worker/checker/distgit.py

```diff
@@ -131,48 +132,77 @@ def pre_check(self) -> bool:
     Report in issue repository if not.
     """
 
     def pre_check(self) -> bool:
-        valid = True
-        msg_to_report = None
+        if self.data.event_type in (NewHotnessUpdateEvent.__name__,):
+            valid = True
+            msg_to_report = None
+
+            if not self.package_config.upstream_project_url:
+                msg_to_report = (
+                    "`upstream_project_url` is not set in "
+                    "the dist-git package configuration."
+                )
+                valid = False
 
-        if not self.package_config.upstream_project_url:
-            msg_to_report = (
-                "`upstream_project_url` is not set in "
-                "the dist-git package configuration."
-            )
-            valid = False
+            if not (
+                self.data.event_dict.get("repo_name")
+                and self.data.event_dict.get("repo_namespace")
+            ):
+                msg_to_report = (
+                    "We were not able to parse repo name or repo namespace from the "
+                    f"upstream_project_url '{self.package_config.upstream_project_url}' "
+                    f"defined in the config."
+                )
+                valid = False
 
-        if not (
-            self.data.event_dict.get("repo_name")
-            and self.data.event_dict.get("repo_namespace")
-        ):
-            msg_to_report = (
-                "We were not able to parse repo name or repo namespace from the "
-                f"upstream_project_url '{self.package_config.upstream_project_url}' "
-                f"defined in the config."
-            )
-            valid = False
-
-        if not self.data.tag_name:
-            msg_to_report = "We were not able to get the upstream tag name."
-            valid = False
-
-        if msg_to_report:
-            logger.debug(msg_to_report)
-            report_in_issue_repository(
-                issue_repository=self.job_config.issue_repository,
-                service_config=self.service_config,
-                title=f"Pull from upstream could not be run for update "
-                f"{self.data.event_dict.get('version')}",
-                message=msg_to_report,
-                comment_to_existing=msg_to_report,
+            if not self.data.tag_name:
+                msg_to_report = "We were not able to get the upstream tag name."
+                valid = False
+
+            if msg_to_report:
+                logger.debug(msg_to_report)
+                report_in_issue_repository(
+                    issue_repository=self.job_config.issue_repository,
+                    service_config=self.service_config,
+                    title=f"Pull from upstream could not be run for update "
+                    f"{self.data.event_dict.get('version')}",
+                    message=msg_to_report,
+                    comment_to_existing=msg_to_report,
+                )
+
+            return valid
```

Comment:
I would say these checks are relevant also when retriggering, WDYT?

therefore, I think what needs to be implemented is getting the last tag since from the PR we do not have any information about the upstream tag

Reply:
Yes, I suppose it makes sense. Where should I implement that though? It can't be in the handler because it has to run before pre-check, right?

Reply:
you're right, it's a little bit tricky as it also doesn't make sense for the PR event itself...so maybe we can allow no `tag_name` in the pre_check if we are handling PR event, and add the getting of the tag in the handler if it is None?